### PR TITLE
OSD-10058: EC2 client and list Instances functionality

### PR DESCRIPTION
**Pull request by ramonbutter**

### CODE REVIEW

**Comment by ramonbutter** [comment]
wow +30k loc. 
Most of it is part of the ec2 interface mock.
Maybe we really have to redefine the ec2 interface only with the functions we need and create our mocks on that.
WDYT?

File: cadctl/cmd/cluster-missing/cluster-missing.go [hunk 1]
```diff
@@ -28,6 +28,22 @@ var ClusterMissingCmd = &cobra.Command{
 	Use:   "cluster-missing",
 	Short: "Will remediate the cluster-missing alert",
 	RunE: func(cmd *cobra.Command, args []string) error {
+		/*
```
Comment (rafael-azevedo) [inline cadctl/cmd/cluster-missing/cluster-missing.go:31]: We don't need to store these comments

**Review by rafael-azevedo — COMMENTED** [review]
It all looks good to me except the commented out section in the main

**Pull request merged**